src/Metal/MTLIntersectionFunctionTable.cs

```diff
@@ -17,17 +17,16 @@ public static void SetBuffers (this IMTLIntersectionFunctionTable table, IMTLBuf
 			if (offsets == null)
 				throw new ArgumentNullException (nameof (offsets));
 
+			var bufferPtrArray = new IntPtr [buffers.Length];
```

Comment:
looks like nice place for a `stackalloc`(but it might need to have a max length check) as you'll be inside an `unsafe` context soonish anyway